kotlinx-coroutines-core/common/src/Annotations.kt

```diff
@@ -90,3 +90,26 @@ public annotation class ObsoleteCoroutinesApi
             "so stable API could be provided instead"
 )
 public annotation class InternalCoroutinesApi
+
+/**
+ * Marks declarations that cannot be safely inherited from.
+ */
+@Target(AnnotationTarget.CLASS)
+@RequiresOptIn(
+    level = RequiresOptIn.Level.WARNING, message =
+    "Inheriting from this kotlinx.coroutines API is unstable. " +
+        "Either new methods may be added in the future, which would break the inheritance, " +
+        "or correctly inheriting from it requires fulfilling contracts that may change in the future."
+)
+public annotation class BrittleForInheritanceCoroutinesApi
```

Comment:
We now have three terms in our glossary -- experimental, brittle and unstable.

I suggest sticking with `experimental` as the most straightforward one -- most of our "Not stable for inheritance" mean "as new methods might be added to this interface in the future"